searchcode/fetchfile.py

```diff
@@ -0,0 +1,87 @@
+import os
+import requests
+
+from .searchcodeapicaller import SearchcodeApiCaller
+
+def work(outputdir, base_url, *, start=0, offset=1, per_page=20, num_limit=0):
+    """
+    Call searchcode api starting from page={start} and with {offset} pages.
+    Stop when result is empty.
+    Only doenload files from github.
+    Download files from master branch, then try develop branch if master branch doesn't exist.
+
+    Downloaded files will be named as {username}_{reponame}_{path_to_file}_{filename}.py
+    Files with the same name will be wiped out.
+
+    Arguments:
+        outputdir:
+            A directory to put all the output file
+        base_url:
+            Searchcode api url.
+
+    Keyword Arguments:
+        start:
+            Starting page. Start from 0.
+        offset:
+            Offset pages. p=start, p=start + offset ...
+        per_page:
+            Number of repository per page (per request).
+        num_limit:
+            Numer of repository to download accross all threads. Set 0 for no limit.
+    """
+    BRANCH = ['master', 'develop']
```

Comment:
Great keyword list. Then, we can also add the frequently used branch tag names such as "dev" and "staging".